crates/tests/implement/tests/identity_from.rs

```diff
@@ -0,0 +1,97 @@
+#![allow(non_snake_case)]
+
+use windows::{core::*, Foundation::*};
+
+#[implement(IStringable, IClosable)]
+struct Test;
+
+impl IStringable_Impl for Test {
+    fn ToString(&self) -> Result<HSTRING> {
+        todo!()
+    }
+}
+
+impl IClosable_Impl for Test {
+    fn Close(&self) -> Result<()> {
+        todo!()
+    }
+}
+
+// This tests that the interface_hierarchy macro correctly implement From<T> and From<&T> for interfaces.
+#[test]
+fn identity_from() -> Result<()> {
+    {
+        let inspectable: IInspectable = Test.into();
+        assert_eq!(
+            inspectable.GetRuntimeClassName()?,
+            "Windows.Foundation.IStringable"
+        );
+
+        let unknown = Into::<&IUnknown>::into(&inspectable);
+        assert_eq!(
+            unknown.cast::<IInspectable>()?.GetRuntimeClassName()?,
+            "Windows.Foundation.IStringable"
+        );
+
+        let unknown = Into::<IUnknown>::into(inspectable);
+        assert_eq!(
+            unknown.cast::<IInspectable>()?.GetRuntimeClassName()?,
+            "Windows.Foundation.IStringable"
+        );
+    }
+    {
+        let stringable: IStringable = Test.into();
+
+        let inspectable = Into::<&IInspectable>::into(&stringable);
+        assert_eq!(
+            inspectable.GetRuntimeClassName()?,
+            "Windows.Foundation.IStringable"
+        );
+
+        let inspectable = Into::<IInspectable>::into(stringable.clone());
```

Comment:
The canonical way I've seen people call typed `From`/`Into` implementations is via `IInspectable::from()`, and the matching `<&IInspectable>::from(&...)`.

Reply:
Well then somebody should write a clippy lint. 😊